src/Reflection/ReflectionClass.php

```diff
@@ -18,34 +20,39 @@ class ReflectionClass implements Reflection
     /**
      * @var NamespaceNode
      */
-    private $declaringNamespace;
+    private $declaringNamespace = null;
 
     /**
      * @var ReflectionMethod[]
      */
-    private $methods;
+    private $methods = [];
 
     /**
      * @var mixed[]
      */
-    private $constants;
+    private $constants = [];
 
     /**
      * @var ReflectionProperty[]
      */
-    private $properties;
+    private $properties = [];
 
     /**
      * @var string
      */
     private $filename;
 
-    private function __construct()
+    public function __construct($className = null)
     {
-        $this->declaringNamespace = null;
-        $this->methods = [];
-        $this->constants = [];
-        $this->properties = [];
+        if ($className) {
```

Comment:
Because that is how `::createFromNode` works.

doing this:

```php
$classInfo = new ReflectionClass('Foo');
```

creates autoload source locator > creates class reflector > calls reflect > locates class > loads source code > loads AST > calls `ReflectionClass::createFromNode()` > which instantiates a new `ReflectionClass` but with no parameters